File: insertIntoMaxTree.py
# https://leetcode.com/problems/maximum-binary-tree-ii/

class Solution(object):
	def insertIntoMaxTree(self, root, val):
		if not root or val > root.val:
			newroot = TreeNode(val)
			newroot.left = root
			return newroot
		
		root.right = self.insertIntoMaxTree(root.right, val)
		return root

File: insertIntoMaxTree.java
// https://leetcode.com/problems/maximum-binary-tree-ii/

class Solution {
	public TreeNode insertIntoMaxTree(TreeNode root, int val) {
		if (root == null || val > root.val) {
			TreeNode newroot = new TreeNode(val);
			newroot.left = root;
			return newroot;
		}
		
		root.right = insertIntoMaxTree(root.right, val);
		return root;
	}
}
